### PR TITLE
[Improvement][Api Module] refactor DataSourceParam and DependentParam, remove spring annotation

### PR DESCRIPTION
**Pull request by wen-hemin**
from #5804 

## Purpose of the pull request

- Remove unnecessary classes in the utils package
- Add UT

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -720,23 +723,50 @@ public ProcessMeta exportProcessMetaData(ProcessDefinition processDefinition) {
     }
 
     /**
-     * correct task param which has datasource or dependent
+     * correct task param which has datasource or dependent for export
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: This method seems to inject some parameters into the TaskNode, I think the comment may be not clear enough.

And if other parameters will be injected in the future, it is better to split this method into two separate method.

Comment (wen-hemin) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: Thx, I'll fix comment.
This is used for import and export process. In the future, if there are inject other parameters, can refactor again.

**Review by ruanwenjun — APPROVED** [review]
+1

**Review by gabry-lab — APPROVED** [review]
+1

**Pull request merged**